SU2_CFD/src/transfer_physics.cpp

```diff
@@ -719,3 +719,133 @@ void CTransfer_SlidingInterface::SetTarget_Variable(CSolver *target_solution, CG
+  Prandtl_Turb      = donor_config->GetPrandtl_Turb();
+  laminar_viscosity = donor_config->GetViscosity_FreeStreamND();
+  cp_fluid          = donor_config->GetSpecificHeat_Fluid();
+  rho_cp_solid      = donor_config->GetSpecificHeat_Solid()*donor_config->GetDensity_Solid();
+
+  PointNormal   = donor_geometry->vertex[Marker_Donor][Vertex_Donor]->GetNormal_Neighbor();
+  Coord         = donor_geometry->node[Point_Donor]->GetCoord();
+  Coord_Normal  = donor_geometry->node[PointNormal]->GetCoord();
+  Normal        = donor_geometry->vertex[Marker_Donor][Vertex_Donor]->GetNormal();
+
+  dist2 = 0.0;
+  Area = 0.0;
+  for (iDim = 0; iDim < nDim; iDim++) {
+    Edge_Vector[iDim] = Coord_Normal[iDim] - Coord[iDim];
+    dist2 += Edge_Vector[iDim]*Edge_Vector[iDim];
+    Area += Normal[iDim]*Normal[iDim];
+  }
+  dist = sqrt(dist2);
+  Area = sqrt(Area);
+
+  /*--- Retrieve temperature solution (later set is as the first donor variable) and its gradient ---*/
+
+  dTdn = 0.0;
+
+  if (compressible_flow) {
+
+    Twall   = donor_solution->node[Point_Donor]->GetPrimitive(0)*Temperature_Ref;
+    Tnormal = donor_solution->node[PointNormal]->GetPrimitive(0)*Temperature_Ref;
+
+    dTdn = (Twall - Tnormal)/dist;
+  }
+  else if (flow || heat_equation) {
+    Twall   = donor_solution->node[Point_Donor]->GetSolution(0)*Temperature_Ref;
+    Tnormal = donor_solution->node[PointNormal]->GetSolution(0)*Temperature_Ref;
+
+//    for (iDim = 0; iDim < nDim; iDim++) {
+//      dTdn += (Twall - Tnormal)/dist * (Edge_Vector[iDim]/dist) * (Normal[iDim]/Area);
+//    }
+
+    dTdn = (Twall - Tnormal)/dist;
+  }
+  else {
+    cout << "WARNING: Transfer of conjugate heat variables is called with non-supported donor solver!" << endl;
+  }
+
+  /*--- Calculate the heat flux density (temperature gradient times thermal conductivity) and set it as second donor variable ---*/
+  if (compressible_flow) {
+
+    iPoint = donor_geometry->vertex[Marker_Donor][Vertex_Donor]->GetNode();
+
+    thermal_conductivityND  = Cp*(laminar_viscosity/Prandtl_Lam);
+    thermal_conductivity    = thermal_conductivityND*donor_config->GetViscosity_Ref();
+
+    heat_flux_density       = thermal_conductivity*dTdn;
+    conductivity_over_dist  = thermal_conductivity/dist;
+  }
+  else if (flow) {
+
+    iPoint = donor_geometry->vertex[Marker_Donor][Vertex_Donor]->GetNode();
+
+    eddy_viscosity          = donor_solution->node[iPoint]->GetEddyViscosity();
+    thermal_conductivityND  = laminar_viscosity/Prandtl_Lam;
+    thermal_conductivity    = thermal_conductivityND*donor_config->GetViscosity_Ref()*cp_fluid;
+
+    heat_flux_density       = thermal_conductivity*dTdn;
+    conductivity_over_dist  = thermal_conductivity/dist;
+  }
+  else {
+
+    thermal_diffusivity     = donor_config->GetThermalDiffusivity_Solid();
+    heat_flux_density       = (thermal_diffusivity*dTdn)*rho_cp_solid;
+    conductivity_over_dist  = thermal_diffusivity*rho_cp_solid/dist;
+  }
+
+  Donor_Variable[0] = Twall;
```

Comment:
To my knowledge (which is going to be a big restriction for this answer),
there is a way to abstract this to a certain point.

Generally, it would be too hard for a driver to figure out the best choises for the CHT couplings. Even worse, there are some ("mathematical") pitfalls that it _had to_ avoid (e. g., assigning only Neumann/heatflux-BC to a solid solver is not well-posed).

However, for some types of interfaces (independent of the complete multi zone problem), there are coupling procedures one can use as a default.
*So far*, I only added the most important compressible and incompressible fluid-solid coupling (I will add solid-solid, ... subsequently soon), for which there seems to be one preferred method which is the only one I committed. (Dirichlet on the flow side, Neumann or Robin on the solid side.)

If one likes to specify other coupling methods, there should be different BC implementations within the solver classes, respectively.

Within the transfer class, one could also use this information to reduce the overhead, as only parts of the data at the interface are needed.
For now, it's only a development comfort for me that I'm always transferring all four informations that might be used by the solver.